### PR TITLE
 Add E2E Tests for Provider Menu

### PR DESCRIPTION
**Pull request by SOHAMPAL23**
# Pull Requests Requirements

- Ensure all significant user interactions are covered.
- Include tests for both positive and negative scenarios.
- Update the test documentation to include new tests.
- Provide instructions on how to run these tests.

## Summary

- **This pull request adds comprehensive end-to-end tests for the Provider Menu. The tests cover all critical user flows, error handling, and ensure the menu's responsiveness and accessibility.**

## Related Issue

https://github.com/DIGI-UW/OpenELIS-Global-2/issues/1681

### CODE REVIEW

File: frontend/cypress/e2e/providerMenu.cy.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    ProviderMenuPage.pageHeading.should("contain.text", "Provider Menu");
+  });
+
+  it("should add a new provider successfully", () => {
+    ProviderMenuPage.openAddModal();
+
+    const providerData = {
+      lastName: "Smith",
+      firstName: "John",
+      telephone: "1234567890",
```
Comment (Bahati308) [inline frontend/cypress/e2e/providerMenu.cy.js]: we could have this info in a separate .json file as well

**Comment by Bahati308** [comment]
the test is not added to the list yet, cypress wont capture it, please add it to the list

**Comment by mozzy11** [comment]
Thanks @SOHAMPAL23 . 
This doesnt llok like it executed in the CI. you need to add it ti the cypress config file

**Comment by SOHAMPAL23** [comment]
Sure @mozzy11 I am working on this as told by @Bahati308  
I will surely commit asap

**Comment by mozzy11** [comment]
Thanks @SOHAMPAL23 . 
Can you see the build failre for provider Menu ??

**Comment by SOHAMPAL23** [comment]
@mozzy11 Yes working on it .

**Review by Bahati308 — CHANGES_REQUESTED** [review]
I realised this page already exists , in our current pages but lacks the test (.cy.js file).It is currently used in modifyOrder.cy.js,  you can go through it and check how it is used then make modifications to it instead of building it from scratch.

File: frontend/cypress/e2e/providerMenu.cy.js (hunk 1 above)
Comment (SOHAMPAL23) [reply frontend/cypress/e2e/providerMenu.cy.js]: @Bahati308 Sure working on that

**Comment by mozzy11** [comment]
Thanks @SOHAMPAL23 .
This is already implemented
